### ISSUE TITLE
Project assigner cannot access projects belonging to organization

### ISSUE THREAD

**Issue by mark-vieira** [elastic/github-actions#6]
I tried setting up the project assigner GitHub action in the `elastic/elasticsearch` repo to assign issues to [this project](https://github.com/orgs/elastic/projects/245) but I ended up getting the following error:

```
Creating a new card for open Issue #567714485 in project [Elasticsearch Build Engineering] column 8061652 mathing label [:Core/Infra/Build], labeled by mark-vieira
##[error]Error adding Issue #567714485 to project Elasticsearch Build Engineering column 8061652: Resource not accessible by integration
```

I think the issue is that the project board in question belongs to the `elastic` or, not the `elastic/elasticsearch` repo. The reason for this is because the board is linked to multiple repos as it's used to coordinate work across more than just a single repo.

It would be nice for the GH action to support such a scenario if at all possible.

**Comment by mark-vieira** [comment]
pinging @alexh97

**Comment by alexh97** [comment]
This is most likely the same problem we're seeing in the Kibana repo. We are working with the Infra team to find a solution: https://github.com/elastic/infra/issues/17904

**Comment by leathekd** [comment]
I responded there.

**Comment by alexh97** [comment]
After much effort by myself and @leathekd, we have concluded that this is a limitation of the current GitHub Actions infrastructure that, unfortunately, has no workaround.  We will need to look for an alternative solution.

**Comment by mark-vieira** [comment]
> After much effort by myself and @leathekd, we have concluded that this is a limitation of the current GitHub Actions infrastructure that, unfortunately, has no workaround. We will need to look for an alternative solution.

Could you provide some more context? I am correct in assuming that the permissions model of GitHub actions is effectively **repo**-centric, and doing cross-repo tasks from actions within a single org is simply not something that's possible from an action?

**Comment by alexh97** [comment]
Actually, now that I think about it, @leathekd may be able to come up with a token that has wide enough permissions to access all repos in the org.  Not sure if that's a great idea from a security standpoint, but I'll let him comment on that.

The problem I was referring to above primarily applies to pull requests, since our model is to have all of our PRs from individual forks, and security tokens are not passed to actions in those scenarios.

**Comment by mark-vieira** [comment]
Right, in the scope of the label assigner, those are always going to be items (issues, prs) that exist on an elastic repository, so I don't think that's necessarily the problem. The problem specifically that I had with that action was that I could not assign an issue labeled in one repo, to a project board that existed at the org level, or perhaps belonged to another repo.

**Comment by alexh97** [comment]
Are you able to do that manually? Especially cross-repo project assignment?

**Comment by mark-vieira** [comment]
Yes. So long as you have permissions to create issues, etc (which I think most folks do, it works fine). For example, I have a project board that exists on the org level, and I have it linked to multiple repositories as I want to track issues in PRs across repos.

**Comment by alexh97** [comment]
OK, let's see if David can come up with a solution for you.  At this point, there's nothing else I can think of doing differently at the action implementation level.

**Comment by mark-vieira** [comment]
Any chance this changes things?

https://github.blog/changelog/2020-05-14-organization-secrets/

**Comment by alexh97** [comment]
It might be worth a try. @leathekd - could you try setting one of those up?

**Comment by schnaustin** [comment]
@leathekd, we're interested in being able to auto-assign issues to org-level project boards within Cloud too - any chance you've had time to look into this issue further?

(CC: @matschaffer)

**Comment by elasticdog** [comment]
I don't believe organization-wide secrets change the equation, but it looks like the use of https://github.com/philschatz/project-bot has passed the vendor assessment [kicked off by @nkammah](https://github.com/elastic/infra/issues/17904#issuecomment-612942132), so that could be a possibility.

**Comment by schnaustin** [comment]
Thanks for the suggestion @elasticdog. I read through the docs for https://github.com/philschatz/project-bot and played around with it in a personal org, unfortunately it doesn't suit our needs.

**Comment by goodroot** [comment]
@richkuz Just put a PR to resolve this -- and it works great, just finished testing it out. 

It adds supports org _and_ user level projects. 

👏 Thanks Rich this is super helpful.

**Comment by richkuz** [comment]
Thanks @goodroot . Here's the mentioned PR that provides org-wide and user-wide project access: https://github.com/elastic/github-actions/pull/36

Example working usage: https://github.com/elastic/app-search-team/blob/master/.github/workflows/project-assigner.yml
Until the PR is merged, feel free to use the action from my fork directly in your workflow: `richkuz/github-actions/project-assigner@v2.1.0`

**Comment by alexh97** [comment]
Thanks @richkuz! I'll try to review that PR by the end of the week.

**Comment by jportner** [comment]
See https://github.com/elastic/github-actions/pull/36#issuecomment-878574390

Merged and released in this repo as [v2.1.0](https://github.com/elastic/github-actions/releases/tag/v2.1.0)!
